File: backend/src/main/java/com/model/employee/Employee.java
package com.model.employee;


import com.model.meeting.Invite;
import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name="employee")
@Data
public class Employee {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "employee_id")
    private int id;

    @Column(name = "office_id")
    private int officeId;

    @Column(name = "employee_name")
    private String name;

    @Column(name = "employee_position")
    private String position;

    @Column(name = "employee_email")
    private String email;

    @OneToMany(mappedBy = "employee", cascade = CascadeType.ALL, orphanRemoval = true)
    List<Invite> invites;

}

File: frontend/src/app/model/employee.ts
export class Employee {
  id: number = 0;
  officeId: number = 0;
  name: string = '';
  position: string = '';
  email: string = '';

  constructor(employee: Employee) {
    this.id = employee.id;
    this.officeId = employee.officeId;
    this.name = employee.name;
    this.position = employee.position;
    this.email = employee.email;
  }

  public toString = () => {
    return this.id + '/' + this.name + '/' + this.officeId + '/' + this.position + '/' + this.email;
  }

}


